src/main/java/com/minecolonies/coremod/colony/requestsystem/resolvers/DeliveryRequestResolver.java

```diff
@@ -47,159 +22,22 @@ public DeliveryRequestResolver(
         super(location, token);
     }
 
-    @Override
-    public TypeToken<? extends Delivery> getRequestType()
-    {
-        return TypeConstants.DELIVERY;
-    }
-
     @Override
     public boolean canResolveRequest(@NotNull final IRequestManager manager, final IRequest<? extends Delivery> requestToCheck)
     {
-        if (manager.getColony().getWorld().isRemote)
-        {
-            return false;
-        }
-
-        final Colony colony = (Colony) manager.getColony();
-        final IWareHouse wareHouse = colony.getBuildingManager().getClosestWarehouseInColony(requestToCheck.getRequester().getLocation().getInDimensionLocation());
-        if (wareHouse == null)
+        final IWareHouse wareHouse = manager.getColony().getBuildingManager().getClosestWarehouseInColony(requestToCheck.getRequest().getStart().getInDimensionLocation());
+        if (wareHouse == null || !wareHouse.getID().equals(getLocation().getInDimensionLocation()))
```

Comment:
This won't work.
Deliveries are not requests solely by a warehouse, but can be requested by any worker.

Reply:
Ye, that just means that the warehouse closest to the delivery position in in charge for a delivery (in particular the deliveryman around that area)

Reply:
Arg it does getClosestWarehouse´

Okey.